pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -159,23 +159,23 @@ private KinesisRecordsBatch getResult(StreamPartitionMsgOffset startOffset, Stre
 
       return new KinesisRecordsBatch(recordList, startShardToSequenceNum.getKey(), isEndOfShard);
     } catch (IllegalStateException e) {
-      LOGGER.warn("Illegal state exception, connection is broken", e);
+      LOGGER.debug("Illegal state exception, connection is broken: {}", e.getMessage());
```

Comment:
I don't think this is what @navina means. IIUC, she was suggesting to log warning for the message only, and log stack trace at debug level.
Side question, do we encounter these exceptions frequently? What is the side effect when such exceptions are hit? Should we consider emitting some metrics for better monitoring?

Reply:
Ok. Yeah that makes sense. And yes some of these can occur frequently e.g. ProvisionedThroughputExceeded or InterruptedException because of timeouts. Let me check what can be done here.